main.go

```diff
@@ -39,10 +65,33 @@ func main() {
 }
 
 func getAllGroups(c *echo.Context) error {
-    res := ResponseTemplate{}
-    groupSlice := []Group{{1, "Foo"}, {2, "Bar"}}
+    var groups Groups
+
+    rC := RedisConnection()
+    defer rC.Close()
+
+    groupKeys, err := rC.Do("KEYS", "group:*")
+    ErrorHandler(err)
 
-    res.Content = groupSlice
+    for _, k := range groupKeys.([]interface{}) {
+        var group Group
+
+        result, err := rC.Do("GET", k.([]byte))
+        ErrorHandler(err) // TODO: handle error here or return error code?
+
+        if err := json.Unmarshal(result.([]byte), &group); err != nil {
+            ErrorHandler(err)
+        }
+        groups = append(groups, group)
+    }
+
+    res := ResponseTemplate{} // TODO: DRY out repetition of setting response values
+    res.Content = groups
+    res.ErrorCode = 0
+    res.ErrorText = "No Error"
+    res.StatusCode = http.StatusOK
+    res.StatusText = "OK"
+    res.Success = true
```

Comment:
In upcoming PR, I opted for defaulting the response template values at the top to the success success values, and in cases where a different HTTP code and text needs to be set, I set them accordingly. Open to alternatives on how to else I could do this.

Began also to extract some logic to own functions like `GetAllGroups` and `SaveGroup` etc where appropriate according to this TODO. :+1: